code/datums/smooth.dm

```diff
@@ -0,0 +1,206 @@
+/atom/var/smooth = 0
+/atom/var/datum/tile_smoother/smoother = null
+/atom/var/list/canSmoothWith=list() // TYPE PATHS I CAN SMOOTH WITH~~~~~
+
+//generic (by snowflake) tile smoothing datum; smooth your icons with this!
+/*
+	Each tile is divided in 4 corners, each corner has an image associated to it; the tile is then overlayed by these 4 images
+	To use this, just set your atom's 'smooth' var to 1. If your atom can't be moved/unanchored, set its 'can_be_unanchored' var to 0.
+	If you don't want your atom's icon to smooth with anything but atoms of the same type, set the list 'canSmoothWith' to null;
+	Otherwise, put all types you want the atom icon to smooth with in 'canSmoothWith' INCLUDING THE TYPE OF THE ATOM ITSELF.
+
+	Each atom has its own icon file with all the possible corner states. See 'smooth_wall.dmi' for a template.
+*/
+/datum/tile_smoother
+	var/image/top_right
+	var/image/top_left
+	var/image/bottom_left
+	var/image/bottom_right
+	var/atom/holder
+	var/cardinal_adj
+	var/list/diagonal_adj
+	var/list/siblings
+	var/movable
+	var/enabled = 1
+
+/datum/tile_smoother/New(var/a_holder, var/list/s_list=null, movable_b = 0)
+	holder = a_holder
+	top_left = image(holder.icon, "1-i")
+	top_right = image(holder.icon, "2-i")
+	bottom_left = image(holder.icon, "3-i")
+	bottom_right = image(holder.icon, "4-i")
+	if(s_list && s_list.len == 0)
+		siblings = null
+	else
+		siblings = s_list
+	movable = movable_b
+
+/datum/tile_smoother/proc/update_adjacencies()
+	cardinal_adj = 0
+	diagonal_adj = list()
+	if(movable)
+		var/atom/movable/A = holder
+		if(!A.anchored)
+			cardinal_adj = 0
+			diagonal_adj = list()
+			return
+		for(var/direction in cardinal)
+			var/atom/movable/AM = find_type_in_direction(holder, direction, siblings)
+			if(AM)
+				if(AM.anchored)
+					cardinal_adj |= direction
+		for(var/direction in diagonals)
+			if(find_type_in_direction(holder, direction, siblings))
+				diagonal_adj |= direction
+	else
+		for(var/direction in cardinal)
+			if(find_type_in_direction(holder, direction, siblings))
+				cardinal_adj |= direction
+		for(var/direction in diagonals)
+			if(find_type_in_direction(holder, direction, siblings))
+				diagonal_adj |= direction
+
+/datum/tile_smoother/proc/smooth()
+	if(!enabled)	return
+	spawn(2)
```

Comment:
I suspect it was related to the singulo causing lots of resmoothing that would then be destroyed the next second